### DISCUSSION TITLE
Disabled all validations

### DISCUSSION THREAD

**Discussion by jbethuel** [firecmsco/firecms#261] [Q&A] [answered]
We are experiencing performance hit because each of the fields are being validate. Is it possible to have an option to disable all validations entirely? And use onPreSave callback handle it? Thanks

![Screenshot 2022-03-18 at 12 18 49 AM](https://user-images.githubusercontent.com/97275020/158846131-56e4d2b2-88a8-40b1-94d4-ad8d710c88de.png)

**Comment by fgatti675** [comment, marked as answer]
Hi @jbethuel 
We need to keep the field validation for enabling the error displays.
We are aware of performance issues in very large forms and are working on virtualising the forms to render only the necessary fields, which should greatly improve performance.
This will be likely included in version 2.0

**Comment by 5hee75** [comment]
@fgatti675 From what I've seen the biggest slowdowns are coming from re-rendering the table cells and a lot of `makeStyles`, not form validations.  For example, even a small form (<10 fields, no arrays or complex objects) can start taking several seconds to open the drawer after just clicking around on a few different rows, but the validations are only run after the drawer opens.  I captured the performance snapshot of that operation and it's spending almost all of its time in `TableCellInternal` doing the `useCellStyles` makeStyles call.  Is that re-render hit also happening because of the form validation performance issues?

![image](https://user-images.githubusercontent.com/24514558/162362416-b77bca45-c039-4f61-8835-53ffecf6704b.png)

**Comment by fgatti675** [comment]
Hi everyone, we are working on multiple performance enhancements for version 2.0
Besides other changes we are also including a schema editor.
You can take a peek at the next version here: https://demo.next.firecms.co/
The code is in the `next` branch
We will release an alpha version when the docs are ready!

**Reply by 5hee75** [reply]
@fgatti675

**UPDATE**

I was able to determine that the drastic slowdowns I was seeing are completely caused by including `React.StrictMode` with the initial app render!

```js
ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById('root')
);
```

Removing the `React.StrictMode` wrapper completely took care of that issue, so now performance matches your example app in the repo.
